### ISSUE TITLE
Add Node Mailer Provider

### ISSUE THREAD

**Issue by ComBarnea** [notifirehq/notifire#24]
### What?
New [Node Mailer](https://github.com/nodemailer/nodemailer) Email Provider

### Why?
To have the ability to use [Node Mailer](https://github.com/nodemailer/nodemailer), smtp as a provider.

### How?
Add new provider based on [this guide](https://docs.notifire.co/docs/community/create-provider)

### Usage
Example for Node Mailer provider registration using smtp, need to be finalized.

```ts
await notifire.registerProvider(
  new NodeMailerProvider({
    from: "support@supercorp.io",
    host: "smtp.test.email",
    port: 587,
    secure: false, // true for 465, false for other ports
    user: "userName", // generated ethereal user
    password: "password",
  })
);
```

**Comment by painotpi** [comment]
Hey, can I pick this?

**Comment by scopsy** [comment]
@painotpi Sure! 

What you can do is copy the SendGrid provider and create a new folder for NodeMailer implementing the `IEmailProvider`.

Attaching a short explanation on our docs https://docs.notifire.co/docs/community/create-provider (needs to be improved but a good start).

**Comment by scopsy** [comment]
@painotpi Just noticed that the `doc` link I gave you is not updated, instead of cloning the template, create a fork of this project `notifire` and create a folder in `providers` and add the new provider there.

Sorry for the confusion

**Comment by painotpi** [comment]
@scopsy No worries, I had started by cloning the template, should I redo it by copying an existing provider folder instead?

**Comment by scopsy** [comment]
@painotpi Yes, it's better to integrate it already to the monorepo. You can just copy the directory you cloned (without the .git file) and add it to the `providers` folder here.

**Comment by ComBarnea** [comment]
Closed by #32.
